src/spaces/compartment.jl

```diff
@@ -0,0 +1,188 @@
+export CompartmentSpace
+
+struct CompartmentSpace{D,F} <: AbstractSpace
+  grid::GridSpace
```

Comment:
GridSpace is type unstable because its type parameters are not stated explicitly as in the definition of the type in `grid.jl`.